### PR TITLE
Solved level6

### PR DESCRIPTION
**Pull request by iuliagatea**

### CODE REVIEW

File: ruby_proj/app/bookstore.rb [hunk 1]
```diff
@@ -21,6 +21,13 @@ def add_to_cart(title)
     add_item_to_cart(cart_item, book) if book.in_stock?
   end
 
+  def remove_from_cart(title)
```
Comment (asarbu) [inline ruby_proj/app/bookstore.rb]: I think we should move some of the logic in the corresponding classes:

1. Cart should know how to delete an item 
2. Book should know how to increment/decrement it's count (I would define these 2 methods and call them everywhere I need)

File: ruby_proj/app/cart.rb [hunk 2]
```diff
@@ -22,7 +22,13 @@ def sub_total
   end
 
   def discount
-    sub_total > 100 ? 0.1 * sub_total : 0
+    if sub_total > 100
+      promotion = 0.1 * sub_total
+      promotion = promotion.to_i if promotion == promotion.to_i
```
Comment (asarbu) [inline ruby_proj/app/cart.rb]: why is this needed ?

File: ruby_proj/level6/data.json [hunk 3]
```diff
@@ -2,14 +2,14 @@
   "books": [
     { "id": 1, "title": "War and Peace", "price": 50, "stock": 9 },
     { "id": 2, "title": "Animal Farm", "price": 30, "stock": 0 },
-    { "id": 3, "title": "Les Misérables", "price": 110, "stock": 1 }
+    { "id": 3, "title": "Les Misérables ", "price": 110, "stock": 1 }
```
Comment (asarbu) [inline ruby_proj/level6/data.json]: not a big issue, but you added an extra space in the title

Comment (iuliagatea) [reply ruby_proj/level6/data.json]: I added an extra space in the title to be like the title of the book on the inferior levels.

File: ruby_proj/app/cart.rb (hunk 2 above)
Comment (iuliagatea) [reply ruby_proj/app/cart.rb]: I have .0 on the promotion and I want to remove it

File: ruby_proj/app/bookstore.rb (hunk 1 above)
Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: you can check it now.

File: ruby_proj/app/bookstore.rb [hunk 4]
```diff
@@ -34,7 +42,7 @@ def add_item_to_cart(cart_item, book)
     else
       cart_item.qty += 1
     end
-    book.stock -= 1 if book.stock
+    book.decrement_stock(1) if book.stock
```
Comment (asarbu) [inline ruby_proj/app/bookstore.rb]: in which case can the `stock` of a book be nil ? I think you should make adjustments so that the value is at least 0 if not higher.

File: ruby_proj/app/bookstore.rb (hunk 1 above)
Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: I can't move `find_item` and `remove_item` in one class. I need to use `find_item` in the add_to_cart method, to check if it already exists in cart.

File: ruby_proj/app/bookstore.rb (hunk 4 above)
Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: Stock is nil for the inferior levels.

File: ruby_proj/app/bookstore.rb (hunk 1 above)
Comment (asarbu) [reply ruby_proj/app/bookstore.rb]: I didn't say to merge them, you could just create a new method with the logic you need.

File: ruby_proj/app/bookstore.rb (hunk 4 above)
Comment (asarbu) [reply ruby_proj/app/bookstore.rb]: would it impact other levels if we default the value to 0 ?

Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: It would impact the `in_stock?` method. I can no longer use this if I pus the stock 0 for levels 1, 2, 3

File: ruby_proj/app/bookstore.rb (hunk 1 above)
Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: ok, but I need the qty of deleted item to be increased in stock. If I merge the methods, I will need to return the qty of the item so the find_item will be called again.

File: ruby_proj/app/bookstore.rb (hunk 4 above)
Comment (asarbu) [reply ruby_proj/app/bookstore.rb]: hmm ok 🤔 . I got it, I guess this is one of those things in which the app wasn't thought to run the same code for all levels

File: ruby_proj/app/bookstore.rb (hunk 1 above)
Comment (asarbu) [reply ruby_proj/app/bookstore.rb]: ooh ok, I got it now, didn't see that you remove the entire quantity of a product. In that case your function could just return the deleted item quantity (if any were deleted) and nil if nothing was deleted

File: ruby_proj/app/book.rb [hunk 5]
```diff
@@ -24,4 +24,12 @@ def in_stock?
     stock.nil? || stock > 0
   end
 
+  def increment_stock(qty)
```
[the remark this answers is not in the document]
Comment (asarbu) [reply ruby_proj/app/book.rb]: I don't see this change reflected in the code 🤔 is everything up-to-date ?

[the remark this answers is not in the document]
Comment (iuliagatea) [reply ruby_proj/app/book.rb]: If a book is added to cart, the decrement_stock method will be called. The increment_stock is called when a customer changes it's mind and then I have to increment with the qty that's in the cart.
